File: elucidate-common-lib/src/test/java/com/digirati/elucidate/common/test/model/annotation/AbstractAnnotationPageTest.java
package com.digirati.elucidate.common.test.model.annotation;

import com.digirati.elucidate.common.model.annotation.AbstractAnnotationPage;

import static org.junit.Assert.assertNotNull;

public class AbstractAnnotationPageTest<T extends AbstractAnnotationPage> extends AbstractObjectTest<T> {

    protected void testAbstractAnnotationPage(T abstractAnnotationPage) {
        assertNotNull(abstractAnnotationPage);
        testAbstractObject(abstractAnnotationPage);
    }
}

File: elucidate-common-lib/src/test/java/com/digirati/elucidate/common/test/model/annotation/AbstractAnnotationPageTest.kt
package com.digirati.elucidate.common.test.model.annotation

import com.digirati.elucidate.common.model.annotation.AbstractAnnotationPage
import org.junit.Assert.assertNotNull

open class AbstractAnnotationPageTest<T : AbstractAnnotationPage?> : AbstractObjectTest<T>() {
    protected fun testAbstractAnnotationPage(abstractAnnotationPage: T) {
        assertNotNull(abstractAnnotationPage)
        testAbstractObject(abstractAnnotationPage)
    }
}